### PR TITLE
case insensitive and make root key for registry

### PR DESCRIPTION
**Pull request by brianbegy**
Creates the concept of a root key for windows registry configs.  This simplifies its use. 
Allow case-insensitive keys.

### CODE REVIEW

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs [hunk 1]
```diff
@@ -185,16 +222,26 @@ public void Set<T>(string key, T value, bool createKeyIfNeeded)
         /// <inheritdocs />
         public void Set<T>(string key, T value)
         {
```
Comment (GerardoBarbitta) [inline src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: Should the Set<T> method also consider key as FullKey(key) as the Get<T> method ?

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs [hunk 2]
```diff
@@ -185,16 +222,26 @@ public void Set<T>(string key, T value, bool createKeyIfNeeded)
...
-                regKey.Close();
-            }
+            this.Set<T>(key, value, false);
         }
 
-        /// <inheritdocs/>
+        /// <inheritdocs />
         public void Delete(string key)
         {
+            this.Delete(key, false);
```
Comment (GerardoBarbitta) [inline src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: Should the Delete method also consider the key as FullKey(key) as the Get method ?

Comment (GerardoBarbitta) [reply src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: ok, I see the other method using isFullyQualified = false

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs [hunk 3]
```diff
@@ -185,16 +222,26 @@ public void Set<T>(string key, T value, bool createKeyIfNeeded)
         /// <inheritdocs />
         public void Set<T>(string key, T value)
         {
-            using (var regKey = this.GetParentKey(key, false, true))
-            {
-                regKey.SetValue(this.GetKeyAsArray(key).Last(), value.ToString());
-                regKey.Close();
-            }
+            this.Set<T>(key, value, false);
```
Comment (GerardoBarbitta) [inline src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: Maybe to add another Set<T> method with the parameter isFullyQualified. To use the FullKey or just the key

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs [hunk 4]
```diff
@@ -172,9 +195,23 @@ public T[] Get<T>(string key, T[] defaultValue, char[] separators, bool errorOnW
...
+        /// <inheritdoc/>
+        public override T Get<T>(string key, T defaultValue, bool errorOnWrongType)
+        {
+            key = FullKey(key);
+            return base.Get<T>(key, defaultValue, errorOnWrongType);
+        }
+
+        public override T Get<T>(string key)
+        {
+            key = FullKey(key);
```
Comment (GerardoBarbitta) [inline src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: thinking here to add a optional parameter as the other methods isFullyQualified. to identify the Get for a FullKey

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs (hunk 1 above)
Comment (brianbegy) [reply src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: yes.  It does.  It calls Set<T>(string key, T value, bool createKeyIfNeeded).  That one first prepends the root key.

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs (hunk 4 above)
Comment (brianbegy) [reply src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: ok done.

File: src/ConfigWrapper/WindowsRegistryConfigWrapper.cs (hunk 3 above)
Comment (brianbegy) [reply src/ConfigWrapper/WindowsRegistryConfigWrapper.cs]: Yeah that's interesting.

**Pull request merged**